[cudnn] Support v8 API in fbcode

xw285cornell opened the pull request:
Summary: It turns out we never turn on cudnn v8 API which blocks bf16 conv. Enable the new v8 API

Test Plan: buck run mode/dev-nosan scripts/xdwang/example:fc_pytorch

Reviewed By: ngimel

Differential Revision: D43784279

xw285cornell commented:
@ngimel @malfet if you can help take a look :)

malfet commented on aten/src/ATen/native/cudnn/Conv_v8.cpp:
```diff
@@ -373,7 +373,7 @@ void generate_and_filter_plans(const cudnnHandle_t handle, cudnn_frontend::Opera
     cudnn_frontend::executionPlans_t new_valid_plans;
     for (auto &plan : valid_plans) {
-      if (plan.getWorkspaceSize() <= max_workspace_size) {
+      if ((size_t) plan.getWorkspaceSize() <= max_workspace_size) {
```
Why this change is needed? If it is, then please use `static_cast<size_t>(plan.getWorkspaceSize())`
```suggestion
      if (plan.getWorkspaceSize() <= max_workspace_size) {
```

ngimel replied:
That's our favorite signed-unsigned comparison probably, can we enable compilation flags to error on it in OSS builds? It's been a nuisance.

ngimel replied:
Given that `getWorkspaceSize()` returns `int64_t` in cudnn frontend, a better fix would be to change our vars like max_worksplace_size and curr_workspace_size to int64_t instead of size_t

xw285cornell replied:
sounds good! I'll make the change and see if I can make this from warning to error in oss :)

malfet reviewed (approved):
LGTM, though I'm surprised why `size_t` cast is needed

xw285cornell commented:
Looks this PR has a conflict with https://github.com/pytorch/pytorch/pull/96723, will probably wait a bit to get in sync

xw285cornell committed:
[cudnn] Support v8 API in fbcode (#96512)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/96512

It turns out we never turn on cudnn v8 API which blocks bf16 conv. Enable the new v8 API

Test Plan: buck run mode/dev-nosan scripts/xdwang/example:fc_pytorch

Reviewed By: ngimel

Differential Revision: D43784279

fbshipit-source-id: 902a4e162807faae874cc9c4baaa90479cd72006
